pkg/apis/core/v1alpha1/types.go

```diff
@@ -256,7 +256,8 @@ type CDIConfig struct {
 
 //CDIConfigSpec defines specification for user configuration
 type CDIConfigSpec struct {
-	UploadProxyURLOverride *string `json:"uploadProxyURLOverride,omitempty"`
+	UploadProxyURLOverride   *string `json:"uploadProxyURLOverride,omitempty"`
+	ScratchSpaceStorageClass *string `json:"scratchSpaceStorageClass,omitempty"`
```

Comment:
I think this should be handled similarly to the upload proxy URL.  There should be a value in the status struct that has whatever the effective value of the scratch space storage class.  It will be either the user provided value in the spec struct or a computed value if the user provided value does not exist.  Clients simply use whatever value is in the status.  Basically move all the logic of determining the scratch space storage class to the config controller.

Reply:
Good point, I will make that change.

Reply:
Updated.